include/flatbuffers/base.h

```diff
@@ -121,6 +121,13 @@
   #define FLATBUFFERS_CONSTEXPR
 #endif
 
+#if (defined(__cplusplus) && __cplusplus >= 201402L) || \
+    (defined(__cpp_constexpr) && __cpp_constexpr >= 201304)
+  #define FLATBUFFERS_CONSTEXPR_CPP14 FLATBUFFERS_CONSTEXPR
```

Comment:
why is this needed separate from `FLATBUFFERS_CONSTEXPR` ?

Reply:
This is because C++11 does not allow `for`-loops in `constexpr`, whereas C++14 does.

By making the definition separate, we can assure the functions become `inline` for C++11, which is what Flatbuffers is targeting, but can be used as `constexpr` when the compile unit is compiled as C++14.

This also why I added the feature check for `__cpp_constexpr >= 201304L` according to the ISO standard, for compilers that might not be fully C++14 compatible, but already expose this feature.

Reply:
ok, makes sense.